src/frontend/src/server/index.ts

```diff
@@ -0,0 +1,29 @@
+import express from "express";
+import csp from "helmet-csp";
+import hsts from "hsts";
+import path from "path";
+import { cspSettings } from "./config/cspSettings";
+
+const buildDir = path.join(process.cwd() + "/build");
+const app = express();
+
+const ONE_YEAR_S = 365 * 24 * 60 * 60;
+
+// Middlewares
+// NOTE(thuang): `app.use()` injection order matters
+app.use(hsts({ maxAge: 2 * ONE_YEAR_S }));
```

Comment:
hsts and csp needs to be added first to Express app middleware chain BEFORE any other controllers, otherwise they won't take effect. 

E.g.,

```ts
// If we add `express.static()` controller first before HSTS and CSP, static file controller will return early before running those two middlewares, thus the response header lacking the security stuff
app.use(express.static(buildDir));

app.use(hsts({ maxAge: 2 * ONE_YEAR_S }));
app.use(csp(cspSettings));

```